nifi-nar-bundles/nifi-kudu-bundle/nifi-kudu-processors/src/main/java/org/apache/nifi/processors/kudu/ScanKudu.java

```diff
@@ -0,0 +1,376 @@
+            .build();
+
+    protected static final Relationship REL_FAILURE = new Relationship.Builder()
+            .name("failure")
+            .description("A FlowFile is routed to this relationship if it cannot be sent to Kudu")
+            .build();
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        final List<PropertyDescriptor> properties = new ArrayList<>();
+        properties.add(KUDU_MASTERS);
+        properties.add(KERBEROS_CREDENTIALS_SERVICE);
+        properties.add(KUDU_OPERATION_TIMEOUT_MS);
+        properties.add(KUDU_KEEP_ALIVE_PERIOD_TIMEOUT_MS);
+        properties.add(TABLE_NAME);
+        properties.add(PREDICATES);
+        properties.add(PROJECTED_COLUMNS);
+        properties.add(BATCH_SIZE);
+        return properties;
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return relationships;
+    }
+
+    static final Set<Relationship> relationships;
+    static {
+        Set<Relationship> rels = new HashSet<>();
+        rels.add(REL_SUCCESS);
+        rels.add(REL_ORIGINAL);
+        rels.add(REL_FAILURE);
+        relationships = Collections.unmodifiableSet(rels);
+    }
+
+    static final String KUDU_TABLE_ATTR = "kudu.table";
+
+    static final String KUDU_ROWS_COUNT_ATTR = "kudu.rows.count";
+
+    protected int batchSize = 500;
+
+    protected KuduTable kuduTable;
+
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) throws LoginException {
+        batchSize = context.getProperty(BATCH_SIZE).evaluateAttributeExpressions().asInteger();
+        createKuduClient(context);
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        final KerberosUser user = getKerberosUser();
+        if (user == null) {
+            trigger(context, session);
+            return;
+        }
+
+        final PrivilegedExceptionAction<Void> privelegedAction = () -> {
+            trigger(context, session);
+            return null;
+        };
+
+        final KerberosAction<Void> action = new KerberosAction<>(user, privelegedAction, getLogger());
+        action.execute();
+    }
+
+    public void trigger(ProcessContext context, ProcessSession session) throws ProcessException {
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        final List<String> projctedColumnNames = Arrays.asList(context.getProperty(PROJECTED_COLUMNS).evaluateAttributeExpressions(flowFile).getValue().split(","));
+        final String tableName = context.getProperty(TABLE_NAME).evaluateAttributeExpressions(flowFile).getValue();
+        if (StringUtils.isBlank(tableName)) {
+            getLogger().error("Table Name is blank or null for {}, transferring to failure", new Object[] {flowFile});
+            session.transfer(session.penalize(flowFile), REL_FAILURE);
```

Comment:
Not sure it makes to penalize the flow file in this case. If the table name is blank or null, it won't change over time and it'd not make sense to have a self-loop failure relationship on the processor, no?